File: python/psfs/library/convnfft.py
import numpy as np

def convnfft(a, b):
    afft = np.fft.fftn(a, axes=(0,1,2))
    bfft = np.fft.fftn(b, axes=(0,1,2), s=a.shape)
    cfft = afft*bfft
    out = np.real(np.fft.ifftn(cfft, axes=(0,1,2)))
    for k in range(len(a.shape)):
        out = np.fft.ifftshift(out, k)

    return out

File: MATLAB/tools/convnfft.m
function out=convnfft(a,b)
afft=fftn(a);
bfft=fftn(b,size(a));
cfft=afft.*bfft;
out=ifftn(cfft);
for k=1:length(size(a)) %dims
    out=ifftshift(out,k);
end

